src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/RMSMappingQuality.java

```diff
@@ -31,32 +32,64 @@
  * </ul>
  *
  */
-public final class RMSMappingQuality extends InfoFieldAnnotation implements StandardAnnotation {
+public final class RMSMappingQuality extends InfoFieldAnnotation implements StandardAnnotation, ReducibleAnnotation {
 
-    public Map<String, Object> annotate(final ReferenceContext ref,
-                                        final VariantContext vc,
-                                        final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap) {
+    @Override
+    public String getRawKeyName() { return GATKVCFConstants.RAW_RMS_MAPPING_QUALITY_KEY;}
+
+    /**
+     * Generate the raw data necessary to calculate the annotation. Raw data is the final endpoint for gVCFs.
+     */
+    @Override
+    public Map<String, Object> annotateRawData(final ReferenceContext ref,
+                                               final VariantContext vc,
+                                               final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap){
         Utils.nonNull(vc);
         if (perReadAlleleLikelihoodMap == null || perReadAlleleLikelihoodMap.isEmpty() ) {
             return null;
         }
 
-        //get all available mapping qualities
-        final List<Integer> qualities = perReadAlleleLikelihoodMap.values().stream().
-                flatMap(pprl -> pprl.getReads().stream().map(read -> read.getMappingQuality()).filter(mq -> mq != QualityUtils.MAPPING_QUALITY_UNAVAILABLE))
-                .collect(Collectors.toList());
-        final double rms = MathUtils.rms(qualities);
-        return Collections.singletonMap(getKeyNames().get(0), formatedValue(rms));
+        final Map<String, Object> annotations = new HashMap<>();
+        final ReducibleAnnotationData<Number> myData = new ReducibleAnnotationData<>(null);
+        calculateRawData(vc, perReadAlleleLikelihoodMap, myData);
+        final String annotationString = formatedValue((double) myData.getAttributeMap().get(Allele.NO_CALL));
+        annotations.put(getRawKeyName(), annotationString);
+        return annotations;
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})//FIXME
+    public void calculateRawData(final VariantContext vc,
+                                 final Map<String, PerReadAlleleLikelihoodMap> pralm,
+                                 final ReducibleAnnotationData rawAnnotations){
+        if (pralm.isEmpty()) {
+            return;
+        }
+
+        //put this as a double, like GATK3.5
+        final double squareSum = pralm.values().stream()
+                .flatMap(likelihoodMap -> likelihoodMap.getReads().stream())
+                .map(GATKRead::getMappingQuality)
+                .filter(mq -> mq != QualityUtils.MAPPING_QUALITY_UNAVAILABLE).mapToDouble(mq -> mq * mq).sum();
```

Comment:
at the 'raw' stage, we just collect the sum of squares. The root part comes in combining.